middleware_retry.go

```diff
@@ -56,8 +58,15 @@ func RetryMiddleware(queue string, mgr *Manager, next JobFunc) JobFunc {
 
 		}()
 
-		err = next(message)
-		if err != nil {
+		switch next(message) {
+		case nil:
+			val, err := message.Get("unique").Bool()
+			if err == nil && val {
+				rc := mgr.opts.client
```

Comment:
I would like to avoid using the Redis client directly at this layer. What are the additional store functions that are needed to support this functionality?

As we stated a few years ago, we are moving in the direction of implementing at least one more additional persistent storage engine to `go-workers2`.